ui/src/dashboards/containers/DashboardPage.tsx

```diff
@@ -138,9 +142,37 @@ class DashboardPage extends Component<Props, State> {
   }
 
   public async componentDidMount() {
-    const {autoRefresh} = this.props
+    const {refreshRate, updateQueryParams} = this.props
+    const {location: {search = ''} = {}} = window
+    const urlSelectedRefresh = search.match(/refresh=([^&#]*)/i)
+    let pollRefreshRate = refreshRate
+    let localStorageRefresh
+
+    if (urlSelectedRefresh) {
+      const option = autoRefreshOptions.find(
+        r => r.label.toLowerCase() === urlSelectedRefresh[1].toLowerCase()
+      )
+      if (option) {
+        this.handleChooseAutoRefresh(option)
+        pollRefreshRate = option.milliseconds
+      } else {
+        localStorageRefresh = autoRefreshOptions.find(
+          r => r.milliseconds === refreshRate
+        )
+      }
+    } else {
+      localStorageRefresh = autoRefreshOptions.find(
+        r => r.milliseconds === refreshRate
+      )
+    }
+
+    if (localStorageRefresh) {
+      updateQueryParams({
+        refresh: localStorageRefresh.label,
```

Comment:
will `localStorageRefresh.label` always exist on `localStorageRefresh`?

Reply:
Yes, because localStorageRefresh is of type `AutoRefreshOption` if it exists (and it does exist here)